lib/ansible/module_utils/basic.py

```diff
@@ -1993,7 +1993,7 @@ def atomic_move(self, src, dest, unsafe_writes=False):
                             try:
                                 try:
                                     out_dest = open(b_dest, 'wb')
-                                    in_src = open(b_src, 'rb')
+                                    in_src = open(b_tmp_dest_name, 'rb')
```

Comment:
OK, now it is clearer. I am sure I want b_tmp_dest_name and not b_src because the exception is raised by os.rename(b_tmp_dest_name, b_dest) line 1987.

So when we catch the exception and try to unsafe_write, we just failed to os-rename b_tmp_dest_name so we must unsafe_write b_tmp_dest_name